Src/Support/Google.Apis.Auth/OAuth2/ComputeCredential.cs

```diff
@@ -92,6 +92,19 @@ public override async Task<bool> RequestAccessTokenAsync(CancellationToken taskC
             return true;
         }
 
+        /// <inheritdoc/>
+        public override async Task<TokenResponse> CreateOidcTokenAsync(string targetAudience, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
we want want two additional settable parameters here:
* format 
* license

  https://cloud.google.com/compute/docs/instances/verifying-instance-identity#request_signature

license=true only works if format=full is set (my 2c is to auto set format=full if the user indicates license=true)

Reply:
Are there other optional parameters coming? If so, we'd probably want to create a new type for that - it's difficult to add extra parameters later.

Reply:
I don't know offhand if other parameters will be set but my guess would be yes (for example, applying user-defined claims)

if it helps, we're addign in vararg options fileds here for the other credential types:

https://github.com/googleapis/google-auth-library-java/blob/5fbaf0f211ca1a1dae05915b224a1ac3c132eb71/oauth2_http/java/com/google/auth/oauth2/IdTokenCredentials.java#L73-L89
-   https://github.com/googleapis/google-auth-library-java/pull/303

---

also for ref, here is the dotnet support for the 'impersonated' credential type; i'll add a comment in that issue pointing out that for impersonated credential, an option is 'includeEmail=true'
https://github.com/googleapis/google-api-dotnet-client/issues/1312

Reply:
I don't see any varargs there - just a builder pattern. I may be missing something. (I definitely don't feel like I've got the complete context of what we're trying to achieve here, but that may just be due to coming back to this two months later.)

Reply:
What i meant is that you can passin arbitrary options to signal the variation of the identity token that the underlying credential type provides.  For example, in java i could pass in the `.setOptions()` field takes in an array of [potential values](https://github.com/googleapis/google-auth-library-java/blob/5fbaf0f211ca1a1dae05915b224a1ac3c132eb71/oauth2_http/java/com/google/auth/oauth2/IdTokenProvider.java#L59) for the `IdTokenCredential` type that will get modified by the true unerlying provider

so in java for compute
```
.setOptions(Arrays.asList(ComputeEngineCredentials.ID_TOKEN_FORMAT_FULL))
```

can also have a value like

or
``` 
.setOptions(Arrays.asList(ComputeEngineCredentials.ID_TOKEN_FORMAT_FULL,
                            ComputeEngineCredentials.LICENSE_TRUE))
```


- interms of why we're getting asking ComputeEngine for these extra values, the instance identity document provides the assertion of extended claims and even license values.  Those can be used by remote system to attest certain things for the origin system. for info on generic id_tokens on gcp, see [this link](https://github.com/salrashid123/salrashid123.github.io/tree/master/google_id_token)


If the pattern we're working with on google-auth-java (eg, appy a  source credential like ComputeENgineCredential into  [IdTokenCredential](https://github.com/googleapis/google-auth-library-java/blob/5fbaf0f211ca1a1dae05915b224a1ac3c132eb71/oauth2_http/java/com/google/auth/oauth2/IdTokenCredentials.java)`  and then derive the token, thats even better

Reply:
Ah, I see - so they're simply on/off options, *not* individual separate values. It still feels like I need a couple of hours to get back into this spec to work out what we should do. I'll see if I can find time later in the week.

Reply:
certainly; pls feel free to ping at anytime; i'd be happy to help where i can.

just note, `google-auth-python` does idTokens different than in `google-auth-java` (as much as i'd like uniformity, that ship has sailed and was also subject to language constructs already in use).

for ref:
`google-auth-python` uses a source credential and _then_ gets an id_token:
```python
from google.auth import compute_engine
creds = compute_engine.IDTokenCredentials( request,
                        target_audience= target_audience)
```
  (as a side note, google-auth-python doesn't even currently use the metadata server to get an id_token...i'm trying to fix that in a different PR [here](https://github.com/googleapis/google-auth-library-python/pull/362) and at the same time apply whatever options i can (at the moment, theyr'e explicitly passed in but as that pr is open, i'll adopt 'options' technique there as well)

in `google-auth-java`, its inverted we get a source credential (eg. `ComputeEngineCredentials`), then init an `IdTokenCredential` and push the source credential to it to derive the id_token

Reply:
Anything i can do to assist here?  

(i'm trying to make a version of this land on our official docs and it'd be good to have dotnet support)
- https://github.com/salrashid123/google_id_token

btw, i'll also verify if the same changes here would work out of the box if we use the credential in grpc calls:
- https://github.com/salrashid123/grpc_google_id_tokens

Reply:
I need to find the time to get back to understanding this, basically. @chrisdunelm may be able to as well, of course.